.github/workflows/codecov.yml

```diff
@@ -0,0 +1,55 @@
+name: Producer C SDK Codecov
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+  pull_request:
+    branches:
+      - develop
+      - master
+
+jobs:
+  linux-gcc-code-coverage:
+    runs-on: ubuntu-20.04
+    env:
+      AWS_KVS_LOG_LEVEL: 2
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
+      - name: Build repository
+        run: |
+          sudo apt install lcov
+          mkdir build && cd build
+          cmake .. -DCODE_COVERAGE=TRUE -DBUILD_TEST=TRUE -DBUILD_COMMON_LWS=TRUE
+          make
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-session-name: ${{ secrets.AWS_ROLE_SESSION_NAME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-duration-seconds: 10800
+      - name: Run tests
+        run: |
+          cd build
+          ulimit -c unlimited -S
+          source ../scripts/get-iot-certs.sh ci
+          timeout --signal=SIGABRT 150m ./tst/producer_test --gtest_break_on_failure
+      - name: Code coverage
+        run: |
+          cd build
+          for test_file in $(find cproducer.dir kvsCommonCurl.dir kvsCommonLws.dir -name '*.gcno'); do gcov $test_file; done
```

Comment:
Was this `gcov` supposed to be changed to `lcov` as well?

Reply:
Good catch. This is not required at all since we are generating with lcov. I had added this when I was debugging issues with lcov for comparison purposes.